src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java

```diff
@@ -227,6 +229,20 @@ else if (!classDataSetAnnotations.isEmpty())
                     throw new IllegalArgumentException(msg);
                 }
             }
+            else if (randomSets > 0)
+            {
+                if (randomSets > iterations.size())
+                {
+                    String msg = MessageFormat.format("Method ''{0}'' is marked to be run with {1} random data sets, but there are only {2}",
+                                                      method.getName(), randomSets, iterations.size());
+                    throw new IllegalArgumentException(msg);
+                }
+                else
+                {
+                    Collections.shuffle(iterations);
```

Comment:
Please try to initialize the shuffle with NeodymiumRandom.getRandom() or pick the elements using NeodymiumRandom on your on. Otherwise we would lose the feature of repeating random executions afterwards.

Reply:
Done